youtube_dl/extractor/porntrex.py

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    urlencode_postdata,
+    ExtractorError,
+)
+
+
+class PornTrexBaseIE(InfoExtractor):
+
+    def _login(self):
+        username, password = self._get_login_info()
+        if username is None:
+            return
+
+        login_page = self._download_webpage(
+            'https://www.porntrex.com/login/', None, 'Downloading login page')
+
+        login_form = self._hidden_inputs(login_page)
+
+        login_form.update({
+            'username': username.encode('utf-8'),
+            'pass': password.encode('utf-8'),
+            'remember_me': str(1).encode('utf-8'),
+        })
+
+        login_page = self._download_webpage(
+            'https://www.porntrex.com/ajax-login/', None,
+            note='Logging in',
+            data=urlencode_postdata(login_form))
+
+        if re.search(r'generic-error hidden', login_page):
+            raise ExtractorError(
+                'Unable to login, incorrect username and/or password',
+                expected=True)
+
+    def _real_initialize(self):
+        self._login()
+
+
+class PornTrexIE(PornTrexBaseIE):
+    _NETRC_MACHINE = 'porntrex'
+    _VALID_URL = r'https?://(?:www\.)?porntrex\.com/video/(?P<id>[0-9]+)/'
+    _TEST = {
+        'url': 'https://www.porntrex.com/video/311136/naomi-gets-fingered-before-the-fucking',
+        'info_dict': {
+            'id': '311136',
+            'ext': 'mp4',
+            'title': 'Naomi gets fingered before the fucking',
+            'uploader': 'cumberland',
+            'description': 'Sexy brunette babe likes to get her tight cunt slammed in hardcore fashion.',
+            'age_limit': 18,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        self.report_extraction(video_id)
+        private_string = 'Only active members can watch private videos.'
+        is_video_private_regex = re.compile(private_string)
+        if re.findall(is_video_private_regex, webpage):
+            self.raise_login_required()
+
+        title = self._html_search_regex(
+            r'<title>(.+?)</title>', webpage, 'title',)
+        movie_urls_regex = re.compile("'(https://www.porntrex.com/get_file/.*?)/'")
```

Comment:
Capturing truncated `https://www.porntrex.com/get_file/` does not make any sense.

Reply:
The media urls appear on the video page in the format `(https://www.porntrex.com/get_file/.*?)/`

Reply:
Do you even read my comments?

Reply:
Hi @dstftw, I have read your comment but am not sure what the concern is. I am not too bright that way.

I need to get the string here to be able to get the media. I have made the re less specific as below but I am not sure that I have addressed your concern so I welcome further advise.
```python
formats = []
movie_urls = re.findall(r"[\"'](https://www.porntrex.com/get_file/.*?)/[\"']", webpage)
for movie_url in movie_urls:
    if re.search(r'screenshots', movie_url):
        continue
    formats.append({'url': movie_url,
                               'height': int(self._search_regex(r'_(\d+)p\.',
                                                                                movie_url,
                                                                                'height',
                                                                                default='480')),
                            })
self._sort_formats(formats)
```